cpp/inc/bond/core/parser.h

```diff
@@ -155,38 +142,18 @@ class StaticParser
 
 
     template <typename T, typename Transform>
-    typename boost::enable_if_c<is_reader<Input>::value && !is_nested_field<T>::value
-                             && !is_fast_path_field<T, Transform>::value, bool>::type
-    NextField(const T&, const Transform& transform)
-    {
-        return transform.Field(T::id, T::metadata, value<typename T::field_type, Input>(_input));
-    }
-
-
-    template <typename T, typename Transform>
-    typename boost::enable_if_c<is_reader<Input>::value && !is_nested_field<T>::value
-                             && is_fast_path_field<T, Transform>::value, bool>::type
+    typename boost::enable_if_c<is_reader<Input>::value && !is_nested_field<T>::value, bool>::type
```

Comment:
Thinking to move the check for `is_nested_field` to utility function as well. Seems the same everywhere.